### ISSUE TITLE
Add reject filter for websockets

### ISSUE THREAD

**Issue by spacemanspiff2007** [openhab/openhab-core#3602]
I'd like to get all events except the `ItemStateEvent` (because with OH4 the `ItemStateUpdateEvent` is available).
Currently the websocket filter works like a whitelist which means I have to know all events and pass a list of them.
However I'd like to receive all events except the `ItemStateEvent` so it would be nice to specify an action in case of a filter match (e.g. "block" or "pass").

**Comment by florian-h05** [comment]
I agree this would be nice to have, but I am currently wondering how this block/pass parameter could be sent.
We only have the payload available for sending this to keep compatibility with events in general ...

**Comment by spacemanspiff2007** [comment]
I don't understand what you are trying to say.

If you look at the [docs](https://www.openhab.org/docs/configuration/websocket.html#filter-by-source-openhab-websocket-filter-source) the filter is created by sending a `WebSocketEvent` with a magic topic.
The easiest way is to add support for a new payload (I don't even think we need to keep the old one for backwards compatibility).
Instead of 
````json
["ItemStateEvent", "ItemStateChangedEvent"]
````
it could be
````json
{
  "allow": ["ItemStateEvent", "ItemStateChangedEvent"],
  "reject": ["ItemStateEvent"]
}
````

Additionally it's not clear to me how I can cancel a filter during runtime.

**Comment by florian-h05** [comment]
> Additionally it's not clear to me how I can cancel a filter during runtime.

Looking at the code, you should be able to cancel all filters by sending an empty array in the payload.

What about specifying reject filters like the following:

```
["!ItemStateEvent"]
```

**Comment by florian-h05** [comment]
> What if I want to only cancel only one filter?

Sorry, I wasn’t clear enough.
You should be able to cancel an individual filter by sending an empty array in its payload, this should work for all filter types.

**Comment by spacemanspiff2007** [comment]
> You should be able to cancel an individual filter by sending an empty array in its payload, this should work for all filter types.

I try to make my question clearer with the following example:
I create the following filter:
`["openhab/items/*/command", "!openhab/items/test/command"]`
How can I remove the `!openhab/items/test/command` while keeping `openhab/items/*/command`
If I send an empty value I'll loose both filters and will be out of sync.

> Your proposal seems a bit over the top to me, more complex to implement than the status quo server-side.

But it would provide a nice filter API which should be generic enough for all websocket activities including logging and can be reused accordingly.
